server/utils/url.js

```diff
@@ -0,0 +1,42 @@
+const Joi = require('joi');
+const { InvalidURL } = require('./coreErrors');
+
+/**
+ * @description Typeof url.
+ * @param {string} str - The url of the backup.
+ * @returns {boolean} Return true for valid url.
+ * @example
+ * isURL();
+ */
+const isURL = (str) => {
+  try {
+    const url = new URL(str);
+    return url.protocol === 'http:' || url.protocol === 'https:';
+  } catch (_) {
+    return false;
+  }
+};
+
+/**
+ * @description Validate the url.
+ * @param {string} url - The url of the backup.
+ * @returns {string} Return a valid url.
+ * @example
+ * validateUrl();
+ */
+function validateUrl(url) {
```

Comment:
You created a generic function "validateUrl" in `utils`, but isn't this behavior very gateway.downloadBackup specific? 

Will this be used elsewhere?

Reply:
What change should I make so that the GET parameter is removed before it’s passed to path.basename()? The validator I wrote, in my opinion, can be used in other places, but it’s largely tied to gateway.downloadBackup.

Reply:
Using `fileWithoutSignedParams` instead ? (as it was before)

```
const fileWithoutSignedParams = fileUrl.split('?')[0];
```